content/v2.1/helpers/form_helper.md

```diff
@@ -0,0 +1,964 @@
+
+```ruby
+<%= form_for("/books", class: "form-horizontal") do |f| %>
+  <div>
+    <%= f.label "books.title" %>
+    <%= f.text_field "books.title", class: "form-control" %>
+  </div>
+
+  <%= f.submit "Create" %>
+<% end %>
+```
+
+This way you get a full control over naming each of the form's element
+
+```html
+<form action="/books" method="POST" accept-charset="utf-8" class="form-horizontal">
+  <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+  <div>
+    <label for="book-title">Title</label>
+    <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+  </div>
+
+  <button type="submit">Create</button>
+</form>
+```
+
+**Method overrides**
+
+While the default HTTP request method is set to `POST`, you can override it by passing in the `method:` keyword.
+
+```ruby
+<%= form_for("/books/123", method: :put) do |f| %>
+ <%= f.text_field "book.title" %>
+ <%= f.submit "Update" %>
+<% end %>
+```
+
+This will override the method giving you the control of the type of action you want to perform.
+
+```ruby
+<form action="/books/123" accept-charset="utf-8" method="POST">
+ <input type="hidden" name="_method" value="PUT">
+ <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+ <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+
+ <button type="submit">Update</button>
+</form>
+```
+
+**Overriding form values**
+
+There is often a scenario where you want to pre-fill the form values with the object loaded from the server. Form helper allows you to override any set of values by simply accepting a hash of attributes you want to replace.
+
+```ruby
+<%= form_for("/songs", values: {song: {title: "Envision"}}) do |f| %>
+ <%= f.text_field "song.title" %>
+ <%= f.submit "Create" %>
+<%= end %>
+```
+
+The value in the `song-title` test field had been filled in from the passed in parameters.
+
+```html
+<form action="/songs" accept-charset="utf-8" method="POST">
+ <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+ <input type="text" name="song[title]" id="song-title" value="Envision">
+
+ <button type="submit">Create</button>
+</form>
+```
+
+## Building blocks
+Below you can see a range of convenient methods for building the fields within an HTML form, integrating with request params and template locals to populate the fields with appropriate values.
+### input
+Generates an input tag without any special handling. For more convenience and other advanced features, see specific field type descriptions in this section.
+
+**Basic usage**
+```ruby
+f.input(type: :text, name: "book[title]", id: "book-title", value: book.title)
+=> <input type="text" name="book[title]" id="book-title" value="Hanami book">
+```
+### csrf_meta_tags
```

Comment:
@swilgosz What if we remove this section in favor of a proper article for CSRF protection?